src/Stack/Build/Source.hs

```diff
@@ -472,32 +504,19 @@ calcFci modTime' fp = liftIO $
             , fciHash = digest
             }
 
-checkComponentsBuildable :: MonadThrow m => [LocalPackage] -> m ()
-checkComponentsBuildable lps =
-    unless (null unbuildable) $ throwM $ SomeTargetsNotBuildable unbuildable
-  where
-    unbuildable =
-        [ (packageName (lpPackage lp), c)
-        | lp <- lps
-        , c <- Set.toList (lpUnbuildable lp)
-        ]
-
 -- | Get 'PackageConfig' for package given its name.
 getPackageConfig :: (MonadIO m, MonadReader env m, HasEnvConfig env)
-  => BuildOptsCLI
-  -> PackageName
-  -> Bool
-  -> Bool
+  => Map FlagName Bool
+  -> [Text]
   -> m PackageConfig
-getPackageConfig boptsCli name isTarget isLocal = do
-  bconfig <- view buildConfigL
+getPackageConfig flags ghcOptions = do
```

Comment:
Although not introduced in this PR `MonadIO` shouldn't be in the type signature. Considering that this function's type signature is being changed in this PR, might as well fix it all the way ;)